src/main/java/seedu/duke/userinterface/Mode.java

```diff
@@ -1,6 +1,56 @@
 package seedu.duke.userinterface;
 
-public enum Mode {
-    NOTEBOOK,
-    TIMETABLE
+import seedu.duke.notebooks.Notebook;
+import seedu.duke.notebooks.NotebookShelf;
+import seedu.duke.notebooks.Section;
+import seedu.duke.tasks.TaskList;
+
+public class Mode {
+    private AppMode appMode;
+    private TaskList taskList;
+    private NotebookShelf currentBookShelf;
```

Comment:
We only ever have one NotebookShelf per instance, so perhaps this should be renamed to `notebookShelf` instead of `currentBookShelf`?

Reply:
Sounds good